x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationUtils.java

```diff
@@ -44,6 +62,33 @@
 
 public final class AuthorizationUtils {
 
+    private static final Logger logger = LogManager.getLogger(AuthorizationUtils.class);
+
+    /**
+     * This map holds parent-child action relationships for which we can optimize authorization
+     * and pre-authorize child actions if the parent action is successfully authorized.
+     * Normally every action would be authorized on a local node on which it's being executed.
+     * Here we define all child actions for which the authorization can be safely skipped
+     * on remote node as they only access a subset of resources.
+     */
+    private static final Map<String, Set<String>> PRE_AUTHORIZED_CHILD_ACTIONS_BY_PARENT = Map.of(
+        "indices:data/read/search",
+        Set.of(
+            "indices:data/read/search[free_context/scroll]",
+            "indices:data/read/search[free_context]",
+            "indices:data/read/search[clear_scroll_contexts]",
+            "indices:data/read/search[phase/dfs]",
+            "indices:data/read/search[phase/query]",
+            "indices:data/read/search[phase/query/id]",
+            "indices:data/read/search[phase/query/scroll]",
+            "indices:data/read/search[phase/query+fetch/scroll]",
+            "indices:data/read/search[phase/fetch/id/scroll]",
+            "indices:data/read/search[phase/fetch/id]",
+            "indices:data/read/search[can_match]",
+            "indices:data/read/search[can_match][n]"
+        )
+    );
```

Comment:
We talked this over and concluded that only the following child actions would be covered by pre-authorization:
indices:data/read/search[free_context]
indices:data/read/search[phase/dfs]
indices:data/read/search[phase/query]
indices:data/read/search[phase/query/id]
indices:data/read/search[phase/fetch/id]
indices:data/read/search[can_match]
indices:data/read/search[can_match][n]

Reply:
Adjusted in https://github.com/elastic/elasticsearch/pull/91886/commits/d51306f9275bb915979e606d718c28b3210d04d0